vcs/src/main/java/org/openjdk/skara/vcs/git/GitRepository.java

```diff
@@ -1720,4 +1720,17 @@ private Diff diffStaged() throws IOException {
             throw t;
         }
     }
+
+    @Override
+    public boolean isFileUpdated(String filename) throws IOException {
```

Comment:
This method seems a bit too specialized to me. It should at least take a Hash argument. I think it would be even better to use the existing `commits` method from which you can get to the Diff and Patch objects. That way we have properly parsed and typed data to find the filename in.